### PR TITLE
CUMULUS-408: HTTPS Ingest

### PR DESCRIPTION
**Pull request by nemreid**
**Summary:** Summary of changes

Addresses [CUMULUS-408: HTTPS Ingest Test](https://bugs.earthdata.nasa.gov/browse/CUMULUS-408)

## Changes

* Fix HTTPS Ingest
* Add new `certificateUri` field to providers to allow custom CA bundles.

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Adhoc testing
- [x] Integration tests

### CODE REVIEW

File: example/spec/helpers/ecsHelpers.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+const { ecs } = require('@cumulus/aws-client/services');
+
+async function findAsyncOperationTaskDefinitionForDeployment(stackName) {
```
Comment (nemreid) [inline example/spec/helpers/ecsHelpers.js:5]: Pagination fix. All updates to Async* spec tests are to use this fix.

File: DATA_MODEL_CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+
+- **CUMULUS-408**
```
Comment (laurenfrederick) [inline DATA_MODEL_CHANGELOG.md:12]: Do we need to make dashboard updates for this?

File: example/spec/helpers/Providers.js [hunk 3]
```diff
@@ -41,17 +41,28 @@ const buildFtpProvider = async (postfix = '') => {
   return provider;
 };
 
-const buildHttpProvider = async (postfix = '') => {
+const fakeProviderPortMap = {
+  http: process.env.PROVIDER_HTTP_PORT ? Number(process.env.PROVIDER_HTTP_PORT) : 3030,
+  https: process.env.PROVIDER_HTTPS_PORT ? Number(process.env.PROVIDER_HTTPS_PORT) : 4040,
+};
+
+const buildHttpOrHttpsProvider = async (postfix = '', protocol = 'http', systemBucket) => {
```
Comment (laurenfrederick) [inline example/spec/helpers/Providers.js:49]: I'm confused here about why defaults are specified for the first 2 params if you have to specify the `systemBucket` param.

Comment (nemreid) [reply example/spec/helpers/Providers.js:49]: Fair point, I can switch them around.

File: DATA_MODEL_CHANGELOG.md (hunk 2 above)
Comment (nemreid) [reply DATA_MODEL_CHANGELOG.md:12]: Looks like the providers page has fields instead of a JSON form, so yes.

Comment (nemreid) [reply DATA_MODEL_CHANGELOG.md:12]: It's an optional field for edge case providers that use self-signed certificates, so it's not exactly top priority, but we should file a ticket.

Comment (nemreid) [reply DATA_MODEL_CHANGELOG.md:12]: Filed CUMULUS-1992 for this.

**Pull request merged**